DNN Platform/DotNetNuke.Abstractions/Portals/PortalAliasCollection.cs

```diff
@@ -0,0 +1,72 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information
+
+namespace DotNetNuke.Abstractions.Portals
+{
+    using System;
+    using System.Collections;
+
+    /// <summary>
+    /// The <see cref="IPortalAliasInfo"/> Collection which provides additional
+    /// helper functions for managing collections of <see cref="IPortalAliasInfo"/>.
+    /// </summary>
+    [Serializable]
+    public class PortalAliasCollection : DictionaryBase
```

Comment:
looking at this implementation, at first glance I don't see any reason to have our own custom collection type

Reply:
@bdukes and I were discussing this here - https://github.com/dnnsoftware/Dnn.Platform/pull/4021#discussion_r480122983

It sounds like the 3 of us agree that this implementation isn't necessary. I still plan to double check to make sure that we aren't overlooking anything.